optimum/intel/ipex/modeling_base.py

```diff
@@ -325,9 +318,11 @@ def __init__(
     ):
         # Perform the initial warmup at the end of __init__
         super().__init__(model, config, model_save_dir=model_save_dir, warmup=False)
+        GenerationMixin.__init__(self)
 
         self.normalized_config = NormalizedConfigManager.get_normalized_config_class(config.model_type)(config)
         self.model_dtype = kwargs.get("model_dtype", self.dtype)
+        self._dtype = self.model_dtype
```

Comment:
I think we should deprecate the `model_dtype` attribute (redundant with `_dtype`), adding a warning that it will be removed after v1.18.0 

```
    @property
    def model_dtype(self):
        # add a warning 
        return self._dtype 
```